### ISSUE TITLE
aria-hidden should not be set to true on modal 

### ISSUE THREAD

**Issue by romans12** [react-bootstrap/react-bootstrap#4723]
## Describe the bug
The modal is currently setting the container to have the attribute of `aria-hidden=true`, but this is not allowed since there are focusable elements inside it.

https://dequeuniversity.com/rules/axe/3.3/aria-hidden-focus?application=axeAPI

## To Reproduce

1. Use axe-core and test Modal

## Reproducible Example
https://react-bootstrap.netlify.com/components/modal

## Environment (please complete the following information)
macOS
Chrome 77
React-Bootstrap 1.0.0-beta.9

**Comment by jordanlapointe** [comment]
Is there a known workaround for this bug? I've been unable to get a modal working as expected with the latest releases (tried betas 12, 13 and 14).

**Comment by malsf21** [comment]
I've been poking around in the source and I'm not exactly sure what's causing it. I'm down to submit a PR though if I can figure it out!

**Comment by jeffredodd** [comment]
This pattern is consistent with the WAI ARIA Best practice documentation for Dialog seen here...
https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/dialog.html

and in more detail here...
https://www.w3.org/TR/wai-aria-practices-1.1/examples/dialog-modal/alertdialog.html

Also for reference this value is set in react-bootstrap/react-overlays [here](https://github.com/react-bootstrap/react-overlays/blob/202c96ba970c87394f245fcef914e94bd8a59a8b/src/utils/manageAriaHidden.js#L18).

**Comment by jquense** [comment]
Yes this is expected and correct way to implement modal behavior for SR's that don't support `aria-modal`

**Comment by jordanlapointe** [comment]
@jeffredodd The document linked does suggest using aria-hidden, but it only suggests to use the property for the inert layer and explicitly recommends not hiding the modal itself:

> In legacy dialog implementations where aria-hidden is used to make content outside a dialog inert for assistive technology users, it is important that:
> 1. aria-hidden is set to true on each element containing a portion of the inert layer.
> 2. **The dialog element is not a descendant of any element that has aria-hidden set to true.**

When testing a modal with VoiceOver on MacOS 10.14, I've found that the modal contents are completely inaccessible and will not be read to the user.

@jquense I believe this ticket is valid and should be reopened.

**Comment by jeffredodd** [comment]
@jordanlapointe Is your modal not existing outside of the React Root in a Portal? Also upon looking I see this issue is also present on the docs site. @jquense I can try to tackle this bug. 🐛. 

I'm guessing its an issue in the Overlays repository.

**Comment by jquense** [comment]
I don't think this is still a bug: https://react-bootstrap.netlify.com/components/modal/ shows that the modal content is not hidden, nor is any of it's ancestors. I think this was fixed upstream already

**Comment by jeffredodd** [comment]
Looks good! Thanks for clarifying @jquense

**Comment by jordanlapointe** [comment]
@jquense Could you validate/sanity check the sandbox example below? It's the same react-bootstrap version and same code as the example in the docs, but it has the aria-hidden bug behavior. The one in the docs works as expected.

https://codesandbox.io/s/code-sandbox-examples-jvf8v

Tested using MacOS Safari and VoiceOver

**Comment by jquense** [comment]
The netlify docs is actually from master so it's not the same as beta 14 we need to push a new release

**Comment by m4r71n** [comment]
Only as note, 0.33.0 is also affected.
Tested on Windows with Firefox and NVDA.
